webapp/app.py

```diff
@@ -3,32 +3,32 @@
 from flask import jsonify, render_template
 
 from webapp import create_app
-from webapp.cache import init_cache
-from webapp.sso import init_sso, login_required
-from webapp.tasks import get_tree_async, init_tasks
+from webapp.models import Webpage, db
+from webapp.sso import login_required
+from webapp.tasks import get_tree_async
 
 app = create_app()
 
-# Initialize SSO
-init_sso(app)
-
-# Initialize cache
-cache = init_cache(app)
-
-# Initialize tasks
-init_tasks(app, cache)
-
 
 @app.route("/get-tree/<string:uri>", methods=["GET"])
 @app.route("/get-tree/<string:uri>/<string:branch>", methods=["GET"])
 @login_required
-def get_tree(uri, branch="main"):
+def get_tree(uri: str, branch="main"):
     # Get tree if already generated, otherwise queue task
     # and return empty array.
     # This prevents concurrent cloning of the same repository.
-    tree = get_tree_async(uri, branch, app, cache)
-
-    response = jsonify({"name": uri, "templates": tree})
+    tree = get_tree_async(uri, branch, app)
+    response_data = {"name": uri, "templates": tree}
+
+    # Return owner id, owner name, stakeholders, and status if webpage exists.
+    webpage = db.session.execute(
+        db.select(Webpage).where(Webpage.url == uri)
+    ).scalar()
+    if webpage:
+        response_data["owner"] = webpage.owner.name
+        response_data["status"] = webpage.status.value
```

Comment:
I didn't get this part. Why do we need to return "owner" and "status" in the get-tree endpoint?

Reply:
ah yes, that's true, but will it add those field to each page in the tree structure?

Reply:
Oh, I get you now.
I'll update it to do this